src/tike/ptycho/ptycho.py

```diff
@@ -174,16 +176,27 @@ def reconstruct(
             logger.info("{} for {:,d} - {:,d} by {:,d} frames for {:,d} "
                         "iterations.".format(algorithm, *data.shape[1:],
                                              num_iter))
-            if split == 'grid':
-                scan, data = split_by_scan_grid(
-                    operator,
-                    pool.num_workers,
-                    scan,
-                    data,
+            # Divide the inputs into regions and mini-batches
+            num_batch = 1
+            if batch_size is not None:
+                num_batch = max(
+                    1,
+                    int(data.shape[1] / batch_size / pool.num_workers),
                 )
-            else:
-                scan, data = split_by_scan_stripes(pool, scan, data,
-                                                   operator.fly)
+            data, scan = split_by_scan_grid(
+                data,
+                scan,
+                (2 if pool.num_workers > 1 else 1, (pool.num_workers + 1) // 2),
+                operator.fly,
```

Comment:
How the fly is determined?

Reply:
If the number of scan positions is greater than the number of diffraction patterns (nframes), then the number of fly-scan positions per frame is nscan/nframes.

Reply:
Get it. I'm fine with the changes. You can merge it whenever you're ready.